### PR TITLE
GITOPS-3617 Create Job from Cronjob feature failing with error cannot create resource "jobs"

### PR DESCRIPTION
**Pull request by chengfang**
**What type of PR is this?**
 /kind bug

**What does this PR do / why we need it**:
Add the necessary batch permissions to the policy rules for argocd server to support creating batch job

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:
https://issues.redhat.com/browse/GITOPS-3617

**How to test changes / Special notes to the reviewer**:
Successfully tested through manual verification in Argo CD UI:
* build the argocd-operator images with the changes in this PR, and push to quay.io/username/argocd-operator
* configure a new CustomCatalog in OpenShift, pointing to the above quay.io images
* in OpenShift OperatorHub, search the modified custom argocd-operator, and install it
* create an Argo CD instance under the installed custom argocd-operator
* navigate to Argo CD UI, and create an Argo CD application that contains a batch job. For instance, https://github.com/chengfang/argocd-example-apps/tree/master/batch
* in the application details page of the above application, try to create a batch job, and it should succeed. See the attached screenshot in [GITOPS-3617](https://issues.redhat.com/browse/GITOPS-3617)
* See the [test result screenshot](https://issues.redhat.com/secure/attachment/13133097/create%20batch%20job%20in%20argocd%20ui.png) attached to the jira.

### CODE REVIEW

File: controllers/argocd/policyrule.go [hunk 1]
```diff
@@ -360,3 +363,18 @@ func appendOpenShiftNonRootSCC(rules []v1.PolicyRule, client client.Client) []v1
...
+		},
+		Resources: []string{
+			"jobs",
+		},
+		Verbs: []string{
+			"create",
+		},
+	}
+	return append(rules, batchPolicyRule)
+}
```
Comment (jaideepr97) [inline controllers/argocd/policyrule.go:380]: I don't think a new function is required for this purpose, we could probably just add this set of permissions as an additional array element in the existing ruleset for the server role and clusterrole directly

Comment (chengfang) [reply controllers/argocd/policyrule.go]: Yeah, appending it inline would be more readable. I've made the change to inline.

**Comment by chengfang** [comment]
the ci job "Build and test the operator / Run e2e tests" failed this time, but they all passed yesterday.

In the log, I'm seeing errors like the following, which looks like a intermittent failure probably due to immature timeout:
```
Error:     case.go:256: resource ArgoCD:test-1-17-custom/example-argocd: .status.phase: value mismatch, expected: 
Available != actual: Pending
```

File: controllers/argocd/policyrule.go [hunk 2]
```diff
@@ -229,6 +240,17 @@ func policyRuleForServerApplicationSourceNamespaces() []v1.PolicyRule {
...
+			APIGroups: []string{
+				"batch",
+			},
+			Resources: []string{
+				"jobs",
+			},
+			Verbs: []string{
+				"create",
+			},
+		},
```
Comment (jaideepr97) [inline controllers/argocd/policyrule.go:253]: @chengfang this block is not required here. This role is specifically meant only to grant permissions to the server  to manipulate applications in non-control plane namespaces

**Review by jaideepr97 — APPROVED** [review]
lgtm, thanks @chengfang

**Review by svghadi — APPROVED** [review]
LGTM. Thanks @chengfang

**Comment by svghadi** [comment]
If we want to backport this, we will have to manually cherry-pick the change. Bot for some reason in creating PR with 0 files changed. - @chengfang

**Pull request merged**